### PR TITLE
4319: Show red asterisk for required p2b fields

### PR DESCRIPTION
**Pull request by holt83**
#### Link to issue

https://platform.dandigbib.org/issues/4319

#### Description

Adds red asterisk to required p2b price fields. See screenshot.

The price field is only required if the event is **not** passive. Therefore we can't use the FAPI #required validation, since [it only cares about the initial field setting in backend](https://www.drupal.org/project/drupal/issues/2855139#comment-12472554).

To still take advangtage of the states API, we can manually theme the required marker in backend and keep our custom validate handler. Some additional logic is required though, since the fields should only be required for existing prices or if no prices currently exist for the event in p2b (we need at least one price for non-passive event for it to work correctly in p2b).

#### Screenshot of the result

4319-p2b-required-fields

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

Also removed the notice about price is not deletable and disabling of the remove button for prices, for new prices that is not created yet:

4319-p2b-unwanted-price-deletable

### CODE REVIEW

File: modules/ding_place2book/ding_place2book.fields.inc [hunk 1]
```diff
@@ -339,13 +365,24 @@ function ding_place2book_widget_validate($form, &$form_state) {
...
+        // exist though (see below).
+        if (isset($price['id']) && !$has_values) {
+          form_set_error("field_place2book][und][place2book][prices_wrapper][prices][$key", t('Both Name and price value is required for prices'));
+        }
+        return $has_values;
+      }, ARRAY_FILTER_USE_BOTH);
+
       if (empty($prices)) {
-        form_set_error('field_place2book][und][place2book][prices_wrapper][prices', t('At least one price should exist.'));
+        form_set_error('field_place2book][und][place2book][prices_wrapper][prices', t('At least one price should exist and botn name and price value is required.'));
```
Comment (JrgenGNielsen) [inline modules/ding_place2book/ding_place2book.fields.inc]: typo: botn

**Comment by holt83** [comment]
@JrgenGNielsen did try a rebuild, but trying again now.

**Comment by holt83** [comment]
@JrgenGNielsen Fixed typo!

**Comment by holt83** [comment]
Tried to rebuild, but still get the same error:

"Did not find Æ in the search result. Looked through 1 pages. (Exception)"

It doesn't appear to be related to the changes in this PR. I can se a lot of the other recent PRs also have this error as well.

**Comment by JrgenGNielsen** [comment]
behat test error does not seem to be related to this issue.
reviewed and approved

**Comment by kasperg** [comment]
@holt83 Can I get you to rerun the build on CircleCI?

**Comment by holt83** [comment]
@kasperg just started a rerun :)

**Pull request merged**